### PR TITLE
IdPMetadata parsing; exception handling; stdout tidyness

### PR DESCRIPTION
**Pull request by fisx**

### CODE REVIEW

**Comment by dkovacevic** [comment]
👍

File: src/Text/XML/DSig.hs [hunk 1]
```diff
@@ -183,9 +194,10 @@ mkSignCredsWithCert mValidSince size = do
 -- list of 'SignCred's.
 {-# NOINLINE verify #-}
 verify :: forall m. (MonadError String m) => NonEmpty SignCreds -> LBS -> String -> m ()
-verify creds el signedID = case unsafePerformIO (verifyIO creds el signedID) of
-  []   -> pure ()
-  errs -> throwError $ show (snd <$> errs)
+verify creds el signedID = case unsafePerformIO (try $ verifyIO creds el signedID) of
```
Comment (neongreen) [inline src/Text/XML/DSig.hs]: By the way, why is it safe to use `unsafePerformIO` here? (Not familiar with what `verifyIO` does)

File: src/Text/XML/DSig.hs [hunk 2]
```diff
@@ -202,11 +214,16 @@ verifyRoot creds el = do
...
 -- *any* succeed, return the empty list.
 verifyIO :: NonEmpty SignCreds -> LBS -> String -> IO [(SignCreds, Either HS.SignatureError ())]
-verifyIO creds el signedID = do
+verifyIO creds el signedID = capture' $ do
   results <- NL.zip creds <$> forM creds (\cred -> verifyIO' cred el signedID)
   case NL.filter (isRight . snd) results of
     (_:_) -> pure []
     []    -> pure $ NL.toList results
+  where
+    capture' :: IO a -> IO a
```
Comment (neongreen) [inline src/Text/XML/DSig.hs]: It would be nicer if there was a comment explicitly complaining about libxml2 being noisy, but ok.

Comment (fisx) [reply src/Text/XML/DSig.hs]: 4c6e74fdef63db24e66fbe88656afe8833c72ab8

File: src/Text/XML/DSig.hs (hunk 1 above)
Comment (fisx) [reply src/Text/XML/DSig.hs]: 4c6e74fdef63db24e66fbe88656afe8833c72ab8

**Pull request merged**